### PR TITLE
Cleanup.

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Agent.Worker/Build/BuildJobExtension.cs [hunk 1]
```diff
@@ -182,7 +182,6 @@ public void ConvertLocalPath(IExecutionContext context, string localPath, out st
                 string cleanRepoText;
                 if(SourceEndpoint.Data.TryGetValue(WellKnownEndpointData.Clean, out cleanRepoText))
                 {
-                    // TODO: expandVariable
```
Comment (ericsciple) [inline src/Agent.Worker/Build/BuildJobExtension.cs]: This is already done in JobRunner but the TODO was never removed.

File: src/Agent.Worker/Build/TeeTFCommandManager.cs [hunk 2]
```diff
@@ -246,17 +245,13 @@ private string FormatArguments(FormatFlags formatFlags, params string[] args)
                 };
                 string arguments = FormatArguments(FormatFlags.None, args);
                 ExecutionContext.Command($@"{_tf} {arguments}");
-                int exitCode = await processInvoker.ExecuteAsync(
+                await processInvoker.ExecuteAsync(
                     workingDirectory: IOUtil.GetWorkPath(HostContext),
                     fileName: _tf,
                     arguments: arguments,
                     environment: null,
+                    requireExitCodeZero: true,
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: @TingluoHuang what do think about this?

File: src/Agent.Worker/TaskManager.cs [hunk 3]
```diff
@@ -135,7 +135,6 @@ public Definition Load(TaskReference task)
             definition.Data = JsonConvert.DeserializeObject<DefinitionData>(json);
 
             // Replace the macros within the handler data sections.
-            // TODO: Do other macros need to be replaced within the handler section? Currently only the $(currentdirectory) macro is handled. Or does the handler need to deal with that.
```
Comment (ericsciple) [inline src/Agent.Worker/TaskManager.cs]: I went ahead and removed this TODO. I *think* it's safe to assume we don't care.

File: src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs [hunk 4]
```diff
@@ -196,4 +227,15 @@ private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
             }
         }
     }
+
+    public sealed class ProcessExitCodeException : Exception
```
Comment (ericsciple) [inline src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs]: I had my eye on GitCommandManager when I added this although I didn't touch that class. I still had a use for the exception in TeeTFCommandManager.

File: src/Agent.Worker/Build/TeeTFCommandManager.cs (hunk 2 above)
Comment (TingluoHuang) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: I am not sure, the only argument i can think of is if the caller want to provide more context to the error message?

File: src/Test/L0/Worker/JobRunnerL0.cs [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
 using Xunit;
 
 namespace Microsoft.VisualStudio.Services.Agent.Tests.Worker
 {
     public class JobRunnerL0
     {
         //// TODO: Fix this test. This test hit file in use issues trying to write to the log file. My guess is it probably has to do with XUnit running tests in parallel.
+        // [Fact]
+        // [Trait("Level", "L0")]
+        // [Trait("Category", "JobRunnerL0")]
```
Comment (ericsciple) [inline src/Test/L0/Worker/JobRunnerL0.cs]: I know this change looks strange. I partially fixed the commented out code :)

I was using this file to test ad-hoc things (e.g. JsonUtility.ToString does not null ref).

File: src/Agent.Worker/Build/TeeTFCommandManager.cs (hunk 2 above)
Comment (ericsciple) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: Then they can. It throws a specific exception type :)

**Pull request merged**